xbmc/filesystem/CurlFile.cpp

```diff
@@ -902,13 +904,13 @@ bool CCurlFile::Download(const std::string& strURL, const std::string& strFileNa
 // Detect whether we are "online" or not! Very simple and dirty!
 bool CCurlFile::IsInternet()
 {
-  CURL url("http://www.msftncsi.com/ncsi.txt");
+  CURL url(StringUtils::Format("http://%s.msftncsi.com/ncsi.txt", g_application.getNetwork().GetFirstConnectedFamily() == AF_INET6 ? "ipv6" : "www"));
```

Comment:
mhhh i think this will only work on properly configured ipv6 interfaces. How sure are we that there are not a dozen missconfigured interfaces which have a valid global ipv6 (or even non global) but no correct route setup? Worst case those will be reported as "no internetconnection available".